frontend/src/routes/mcp-settings.tsx

```diff
@@ -18,11 +18,15 @@ function MCPSettingsScreen() {
   const { data: settings, isLoading } = useSettings();
   const { mutate: saveSettings, isPending } = useSaveSettings();
 
-  const [mcpConfig, setMcpConfig] = useState<MCPConfig | undefined>(
-    settings?.MCP_CONFIG,
-  );
+  const [mcpConfig, setMcpConfig] = useState<MCPConfig | undefined>(undefined);
```

Comment:
```suggestion
  const [mcpConfig, setMcpConfig] = useState<MCPConfig>();
```

When you use `useState<MCPConfig>()` without providing an initial value, it implicitly sets the state to undefined.

Both are corrects solutions, but this one feels more compact 😄